### PR TITLE
[3621] Add MockWebServer for UI tests

### LINKED ISSUE #3621: Mock web server for UI tests
### Context & proposal

We need to create a mock web server for our UI tests.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] The server is capable of returning mocked responses via REST
- [ ] The server is capable of sending mocked events via WS
- [ ] A sample UI test is created to showcase the usage of mocked endpoints

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Closes #3621 

Create a sample UI test using `MockWebServer`.

### 🎨 UI Changes

https://user-images.githubusercontent.com/9600921/172287530-d56927f4-3c6f-40e3-a921-cd6e55f499b2.mp4

### 🧪 Testing

1. Run the `UiTestsSuite` suite to execute all available UI tests
2. Ensure that all the tests complete successfully

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 1]
```diff
@@ -2472,11 +2471,16 @@ internal constructor(
                 )
             }
```
Comment (bychkovdmitry) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:2473]: Don't want to bother with self-signed certificates. That's why using unencrypted protocols here.

**Review by skydoves — APPROVED** [review]
Overall looks good to me :)

**Review by TGazica — APPROVED** [review]
Looks good to me.

**Review by amitkma — APPROVED** [review]
LGTM from core side.

**Pull request merged**